sql/src/main/java/org/apache/apex/malhar/sql/codegen/BeanClassGenerator.java

```diff
@@ -0,0 +1,796 @@
+    toStringNode.instructions.add(new VarInsnNode(Opcodes.ALOAD, 1));
+
+    for (int i = 0; i < fieldList.size(); i++) {
+      TupleSchemaRegistry.SQLFieldInfo info = fieldList.get(i);
+      String fieldName = info.getColumnName();
+      String fieldType = info.getType().getJavaType().getName();
+      String fieldJavaType = getJavaType(fieldType);
+
+      if (i != 0) {
+        toStringNode.instructions.add(new LdcInsnNode(", "));
+        toStringNode.instructions.add(new MethodInsnNode(Opcodes.INVOKEVIRTUAL, "java/lang/StringBuilder", "append",
+            "(Ljava/lang/String;)Ljava/lang/StringBuilder;", false));
+      }
+
+      toStringNode.instructions.add(new LdcInsnNode(fieldName + "="));
+      toStringNode.instructions.add(new MethodInsnNode(Opcodes.INVOKEVIRTUAL, "java/lang/StringBuilder", "append",
+          "(Ljava/lang/String;)Ljava/lang/StringBuilder;", false));
+      toStringNode.instructions.add(new VarInsnNode(Opcodes.ALOAD, 0));
+      toStringNode.instructions.add(new FieldInsnNode(Opcodes.GETFIELD, classNode.name, fieldName, fieldJavaType));
+
+      // There is no StringBuilder.append method for short and byte. It takes it as int.
+      if (fieldJavaType.equals("S") || fieldJavaType.equals("B")) {
+        fieldJavaType = "I";
+      }
+
+      Character pchar = PRIMITIVE_TYPES.get(fieldType);
+      if (pchar == null) {
+        // It's not a primitive type. StringBuilder.append method signature takes Object type.
+        fieldJavaType = "Ljava/lang/Object;";
+      }
+
+      toStringNode.instructions.add(new MethodInsnNode(Opcodes.INVOKEVIRTUAL, "java/lang/StringBuilder", "append",
+          "(" + fieldJavaType + ")Ljava/lang/StringBuilder;", false));
+    }
+
+    toStringNode.instructions.add(new LdcInsnNode("}"));
+    toStringNode.instructions.add(new MethodInsnNode(Opcodes.INVOKEVIRTUAL, "java/lang/StringBuilder", "append",
+        "(Ljava/lang/String;)Ljava/lang/StringBuilder;", false));
+
+    toStringNode.instructions.add(new InsnNode(Opcodes.POP));
+    toStringNode.instructions.add(new VarInsnNode(Opcodes.ALOAD, 1));
+    toStringNode.instructions.add(new MethodInsnNode(Opcodes.INVOKEVIRTUAL, "java/lang/StringBuilder", "toString",
+        "()Ljava/lang/String;", false));
+    toStringNode.instructions.add(new InsnNode(Opcodes.ARETURN));
+
+    classNode.methods.add(toStringNode);
+  }
+
+  /**
+   * This will add a hashCode method for class being generated. <br>
+   * Algorithm is as follows: <br>
+   * <i><p>
+   * int hashCode = 7;
+   * for (field: all fields) {
+   * hashCode = 23 * hashCode + field.hashCode()
+   * }
+   * </p></i>
+   * <br>
+   * <b> For primitive field, hashcode implemenented is similar to the one present in its wrapper class. </b>
+   *
+   * @param classNode
+   * @param fieldList
+   * @throws JSONException
+   */
+  @SuppressWarnings("unchecked")
+  private static void addHashCodeMethod(ClassNode classNode, List<TupleSchemaRegistry.SQLFieldInfo> fieldList)
```

Comment:
I would prefer bytecode version to be 1.7 so you can use Objects.hash function which is easier for bytecode here. And I believe most hadoop distribution is using 1.7+ by default.